packages/generators/generators/lib/plops/api.js

```diff
@@ -3,63 +3,75 @@
-    prompts: [
-      {
-        type: 'input',
-        name: 'id',
-        message: 'API name',
-        validate: input => validateInput(input),
-      },
-      {
-        type: 'confirm',
-        name: 'isPluginApi',
-        message: 'Is this API for a plugin?',
-      },
-      {
-        when: answers => answers.isPluginApi,
-        type: 'list',
-        name: 'plugin',
-        message: 'Plugin name',
-        async choices() {
-          const pluginsPath = join(plop.getDestBasePath(), 'plugins');
-          const exists = await fs.pathExists(pluginsPath);
+    async prompts(inquirer) {
+      const api = await inquirer.prompt([
+        {
+          type: 'input',
+          name: 'id',
+          message: 'API name',
+          validate: input => validateInput(input),
+        },
+        {
+          type: 'confirm',
+          name: 'isPluginApi',
+          message: 'Is this API for a plugin?',
+        },
+        {
+          when: answers => answers.isPluginApi,
+          type: 'list',
+          name: 'plugin',
+          message: 'Plugin name',
+          async choices() {
+            const pluginsPath = join(plop.getDestBasePath(), 'plugins');
+            const exists = await fs.pathExists(pluginsPath);
 
-          if (!exists) {
-            throw Error('Couldn\'t find a "plugins" directory');
-          }
+            if (!exists) {
+              throw Error('Couldn\'t find a "plugins" directory');
+            }
 
-          const pluginsDir = await fs.readdir(pluginsPath, { withFileTypes: true });
-          const pluginsDirContent = pluginsDir.filter(fd => fd.isDirectory());
+            const pluginsDir = await fs.readdir(pluginsPath, { withFileTypes: true });
+            const pluginsDirContent = pluginsDir.filter(fd => fd.isDirectory());
 
-          if (pluginsDirContent.length === 0) {
-            throw Error('The "plugins" directory is empty');
-          }
+            if (pluginsDirContent.length === 0) {
+              throw Error('The "plugins" directory is empty');
+            }
 
-          return pluginsDirContent;
+            return pluginsDirContent;
+          },
+        },
+        {
+          type: 'confirm',
+          name: 'createContentType',
+          default: true,
+          message: 'Create a content-type?',
         },
-      },
-      {
-        type: 'list',
-        name: 'kind',
-        message: 'Please choose the model type',
-        default: 'collectionType',
-        choices: [
-          { name: 'Collection Type', value: 'collectionType' },
-          { name: 'Singe Type', value: 'singleType' },
-        ],
-      },
-      {
-        type: 'confirm',
-        name: 'useDraftAndPublish',
-        default: false,
-        message: 'Use draft and publish?',
-      },
-    ],
+      ]);
+
+      if (!api.createContentType) {
+        return api;
+      }
+
+      return {
+        ...api,
+        ...(await inquirer.prompt([
+          ...getCtNamesPrompts,
```

Comment:
can we either make them all function to be actual getter or remove the `get` prefix as they are just arrays ?